discovery/dnsdisc/client.nim

```diff
@@ -29,7 +29,7 @@ type
   
   ## A Resolver proc takes a DNS domain as argument and
   ## returns the TXT record at that domain
-  Resolver* = proc(domain: string): Future[string] {.gcsafe.}
+  Resolver* = proc(domain: string): Future[string] {.gcsafe, raises: [].}
```

Comment:
:+1: 

callbacks that return future must always be `{.async.}` or equivalently `{.gcsafe, raises: [].}` - the former is preferred but the latter also works